cmd/wego/app/add/cmd.go

```diff
@@ -84,15 +84,33 @@ func runCmd(cmd *cobra.Command, args []string) error {
 	osysClient := osys.New()
 
 	token, err := osysClient.GetGitProviderToken()
-	if err != nil {
-		return err
+
+	if err == osys.ErrNoGitProviderTokenSet {
+		// No provider token set, we need to do the auth flow.
+		url := params.Url
+		if url == "" {
+			// Find the url using an unauthenticated git client. We just need to read the URL.
+			// params.Dir must be defined here because we already checked for it above.
+			url, err = git.New(nil).GetRemoteUrl(params.Dir, "origin")
+			if err != nil {
+				return fmt.Errorf("could not get remote url for directory %s: %w", params.Dir, err)
+			}
+		}
+		// DoAppRepoCLIAuth will take over the CLI and block until the flow is complete.
+		token, err = app.DoAppRepoCLIAuth(url, os.Stdout)
```

Comment:
This is really a nit (and it might be unlikely that we would want to mock this particular case) but the `osys` client has a `osysClient.Stdout()` function that could be mocked so I've tried to use that everywhere except the main function of a cli command just in case we want to swap it out :man_shrugging:

Reply:
Took a look and that call returns a `*File`, which I guess implements the `io.Writer`, so it compiles but seems weird. I'll go with your suggestion so it doesn't bite someone trying to test.